### PR TITLE
[gym] add options to configure or disable xcpretty

### PR DESCRIPTION
**Pull request by bgannin**
Fix [gym] Option to configure/disable xcpretty #5035

Includes the following:
- _-h_ to disable, includes a spec to verify it (bool)
- _-v_ to specify a custom formatter (string)
- _-t_ to use 'test' format instead of 'simple' format (bool)
- _-1_ to generate XML (JUnit) report at path (string)
- _-2_ to generate simple HTML report at path (string)
- _-3_ to generate JSON database at path (string)

### CODE REVIEW

File: gym/lib/gym/options.rb [hunk 1]
```diff
@@ -195,7 +195,51 @@ def self.plain_options
...
+        FastlaneCore::ConfigItem.new(key: :xcpretty_formatter,
+                                     short_option: "-v",
+                                     env_name: "XCPRETTY_FORMATTER",
+                                     description: "A custom xcpretty formatter to use",
+                                     optional: true,
+                                     verify_block: proc do |value|
+                                       UI.user_error!("Formatter file not found at path '#{File.expand_path(value)}'") unless File.exist?(value)
+                                     end),
+        FastlaneCore::ConfigItem.new(key: :xcpretty_report_junit,
+                                     short_option: "-1",
```
Comment (ohwutup) [inline gym/lib/gym/options.rb]: Is there a reason why the short options are set to be integers?

**Comment by ohwutup** [comment]
Thanks for this @bgannin! Great to have some more options here! Just one question for now about the short options above, let me know what you think! 🐙 !

File: gym/lib/gym/options.rb (hunk 1 above)
Comment (bgannin) [inline gym/lib/gym/options.rb]: Ran out of lowercase letters and short_option's length was required as one. Also made them quite distinct, heh. Guess we could go upper case…

Comment (ohwutup) [inline gym/lib/gym/options.rb]: Hey @bgannin! I think we should just remove the `short_option`s all together. They are not required by fastlane, and we should really try to keep them only for _very_ commonly used settings so as to not bring about collisions.
🚀

Comment (bgannin) [inline gym/lib/gym/options.rb]: Done deal 👍

**Comment by ohwutup** [comment]
🚀 Thanks again for this and working through those changes @bgannin !! Going to merge now!

**Pull request merged**